src/main/java/edu/ie3/datamodel/io/source/sql/SqlDataSource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source.sql;
+
+import edu.ie3.datamodel.exceptions.InvalidColumnNameException;
+import edu.ie3.datamodel.io.connectors.SqlConnector;
+import edu.ie3.util.StringUtils;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SqlDataSource<T> {
+
+  protected static final Logger log = LoggerFactory.getLogger(SqlDataSource.class);
+
+  private final SqlConnector connector;
+
+  public SqlDataSource(SqlConnector connector) {
```

Comment:
Code Smell: Constructor of abstract classes should be protected as only their sublasses can be instantiated.